### PR TITLE
fix(frontend): update navbar component to match new format

### LINKED ISSUE #1781: Property 'brand' does not exist on type IntrinsicAttributes
## 💥 Regression Report

Seems to be an [incompatibility](https://github.com/HospitalRun/hospitalrun-frontend/pull/1776/checks?check_run_id=418840720) with new version 0.32.0 of components 

## Last working version

Worked up to version: 0.31.0

Stopped working in version: [0.32.0](https://github.com/HospitalRun/components/pull/228)

```js
TypeScript error in /home/travis/build/HospitalRun/hospitalrun-frontend/src/components/Navbar.tsx(12,7):

Type '{ brand: { label: string; onClick: () => void; }; search: { onClickButton: () => void; onChangeInput: () => void; placeholderText: string; buttonText: string; }; navLinks: { label: string; onClick: () => void; children: { ...; }[]; }[]; }' is not assignable to type 'IntrinsicAttributes & Pick<Props, "ref" | "className" | "children" | "navItems" | "key"> & Partial<Pick<Props, "bg" | "variant">> & Partial<...>'.

Property 'brand' does not exist on type 'IntrinsicAttributes & Pick<Props, "ref" | "className" | "children" | "navItems" | "key"> & Partial<Pick<Props, "bg" | "variant">> & Partial<...>'.  TS2322    
10 |   return (   
11 |     <HospitalRunNavbar  > 
12 |       brand={{       |       ^    
13 |         label: 'HospitalRun',    
14 |         onClick: () => {    
15 |           history.push('/')
```

### PR DESCRIPTION
**Pull request by ocBruno**
BREAKING CHANGE: update components package changing some interfaces

fix #1781

**Changes proposed in this pull request:**

- update components package
- update navbar to match changes

https://github.com/HospitalRun/components
32.2

### CODE REVIEW

**Comment by ocBruno** [comment]
Hey @jackcmeyer 
To be able to finish resolving this efficiently, I need to first create a pull request to update the components library to add className identifiers to the navitems to be able to properly test. 
I will create a pull request for that before finishing up this update.

**Comment by jackcmeyer** [comment]
> Hey @jackcmeyer
> To be able to finish resolving this efficiently, I need to first create a pull request to update the components library to add className identifiers to the navitems to be able to properly test.
> I will create a pull request for that before finishing up this update.

@ocBruno doesn't NavItems already have a `className` https://github.com/HospitalRun/components/blob/9a492170d88d103b7d00454392cd77923ce2f2d3/src/components/Navbar/interfaces.tsx#L4?

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -9,36 +9,36 @@ const Navbar = () => {
...
+          label: 'HospitalRun',
           onClick: () => {
-            // no oop
+            history.push('/')
           },
+        },
+        {
+          type: 'link-list',
+          label: t('patients.label'),
+          onClick: () => undefined,
```
Comment (jackcmeyer) [inline src/components/Navbar.tsx]: `NavLink`'s `onClick` is not required, so we don't need to define it as undefined.

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -47,27 +47,38 @@ const Navbar = () => {
           ],
         },
         {
+          type: 'link-list',
           label: t('scheduling.label'),
-          onClick: () => {
-            // no oop
-          },
+          onClick: () => undefined,
```
Comment (jackcmeyer) [inline src/components/Navbar.tsx]: NavLink's onClick is not required, so we don't need to define it as undefined.

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -9,36 +9,36 @@ const Navbar = () => {
...
```
Comment (jackcmeyer) [inline src/components/Navbar.tsx]: Clicking on the icon should navigate to `/` as well

**Comment by ocBruno** [comment]
> > Hey @jackcmeyer
> > To be able to finish resolving this efficiently, I need to first create a pull request to update the components library to add className identifiers to the navitems to be able to properly test.
> > I will create a pull request for that before finishing up this update.
> 
> @ocBruno doesn't NavItems already have a `className` https://github.com/HospitalRun/components/blob/9a492170d88d103b7d00454392cd77923ce2f2d3/src/components/Navbar/interfaces.tsx#L4?

Ahh yes, I was going to add a default identifiers such as nav-header and nav-icon but now that you mentioned the built in one i'm not sure we need to, I will add classes to each item and test them with those.

**Comment by ocBruno** [comment]
> > Hey @jackcmeyer
> > To be able to finish resolving this efficiently, I need to first create a pull request to update the components library to add className identifiers to the navitems to be able to properly test.
> > I will create a pull request for that before finishing up this update.
> 
> @ocBruno doesn't NavItems already have a `className` https://github.com/HospitalRun/components/blob/9a492170d88d103b7d00454392cd77923ce2f2d3/src/components/Navbar/interfaces.tsx#L4?

Hey @jackcmeyer,

finished updating the tests and suggested a new structure with some improvements we could apply to other tests!

**Pull request merged**